src/canisters/frontend/ic-certified-assets/src/http.rs

```diff
@@ -80,6 +80,39 @@ impl HttpRequest {
             .iter()
             .find_map(|(k, v)| k.eq_ignore_ascii_case(header_key).then_some(v))
     }
+
+    pub fn redirect_from_raw_to_certified_domain(&self) -> HttpResponse {
+        let location = format!(
+            "https://{canister_id}.ic0.app{path}",
+            canister_id = self.get_canister_id(),
+            path = self.url
+        );
+        HttpResponse::build_redirect(308, location)
+    }
+
+    pub fn get_canister_id(&self) -> &str {
+        if let Some(host_header) = self.get_header_value("Host") {
+            if host_header.contains(".localhost") || host_header.contains(".app") {
+                return host_header.split('.').next().unwrap();
+            } else {
+                if let Some(t) = self.url.split("canisterId=").nth(1) {
+                    let x = t.split_once('&');
+                    if let Some(c) = x {
+                        return c.0;
+                    }
+                }
+            }
+        }
+        "canister-id-is-unreachable"
+    }
```

Comment:
I think this can be replaced with https://docs.rs/ic-cdk/latest/ic_cdk/api/fn.id.html

Reply:
good idea, however as discussed I don't know how to make it work. 
the following changes 
![image](https://user-images.githubusercontent.com/21069150/207934978-4588f8d1-6d78-4c20-82f2-fe8145194b7b.png)
make it not possible to run `cargo test` due to compile linking error 
```
cargo test --benches --tests --all-features 
   Compiling ic-certified-assets v0.2.5 (/src/canisters/frontend/ic-certified-assets)
error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "-arch" "arm64" "/var/folders/qr/2sz_4fyx75z7smnl9k554p000000gn/T/rustcSNu69F/symbols.o" 
...  
= note: Undefined symbols for architecture arm64:
            "_canister_self_copy", referenced from:
                ic_cdk::api::id::h072c680cecf25c11 in libic_cdk-f3b53b5d1a916317.rlib(ic_cdk-f3b53b5d1a916317.ic_cdk.8608ab03-cgu.9.rcgu.o)
            "_canister_self_size", referenced from:
                ic_cdk::api::id::h072c680cecf25c11 in libic_cdk-f3b53b5d1a916317.rlib(ic_cdk-f3b53b5d1a916317.ic_cdk.8608ab03-cgu.9.rcgu.o)
          ld: symbol(s) not found for architecture arm64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
          
error: could not compile `ic-certified-assets` due to previous error
```

maybe @ericswanson-dfinity will have an idea?

Reply:
No, but @adamspofford-dfinity did.

> It was just outdated ic0. `cargo update -p ic0`
> 0.18.6 added back the shim for compiling on desktop